src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/magical/InfusedHopper.java

```diff
@@ -104,7 +102,10 @@ public boolean isSynchronized() {
     private boolean isValidItem(@Nonnull Location l, @Nonnull Entity entity) {
         if (entity instanceof Item && entity.isValid()) {
             Item item = (Item) entity;
-            return !SlimefunUtils.hasNoPickupFlag(item) && item.getLocation().distanceSquared(l) > 0.25;
+            // Check if the item cannot be picked up or has the "no pickup" metadata
+            return item.getPickupDelay() != Short.MAX_VALUE
```

Comment:
```suggestion
            return item.getPickupDelay() <= 0
```

getPickupDelay() will count down, so we should respect the pickup delay fully here tbh.

Reply:
No, pickup delay does not count down if it's at MAX_VALUE. This is exactly why it can never be picked up

I can change this though if you wish

Reply:
I know, but for any other value it will.
Tbh I thought we checked for the pickup delay anyway (we do this with the Infused Magnet too)